gammapy/background/off_data_background_maker.py

```diff
@@ -0,0 +1,248 @@
+            Observation selection method.
+        n_obs_max : int, None
+            Maximum number of observations (useful for quick testing)
+        """
+        if selection == 'offplane':
+            obs_table = self.data_store.obs_table[:n_obs_max]
+            min_glat = 5
+            mask = np.abs(obs_table['GLAT']) > min_glat
+            obs_table = obs_table[mask]
+            obs_table = obs_table[['OBS_ID']]
+        elif selection == 'debug':
+            obs_table = self.data_store.obs_table[:n_obs_max]
+            obs_table = obs_table[['OBS_ID']]
+        elif selection == 'old-hap-hd':
+            hessroot = os.environ['HESSROOT']
+            filename = Path(hessroot) / 'hddst/scripts/lookups/lists/acceptance_runs.csv'
+            log.info('Reading {}'.format(filename))
+            obs_hess = Table.read(str(filename), format='ascii')
+            obs_hess = obs_hess['Run'].data
+
+            filename = self.run_list
+            log.info('Reading {}'.format(filename))
+            obs_export = Table.read(str(filename), format='ascii')
+            obs_export = obs_export['col1'].data
+
+            obs_common = sorted(set(obs_hess) & set(obs_export))
+
+            log.info('Runs HESS:   {:6d}'.format(len(obs_hess)))
+            log.info('Runs export: {:6d}'.format(len(obs_export)))
+            log.info('Runs common: {:6d}'.format(len(obs_common)))
+
+            if n_obs_max:
+                log.info('Applying max. obs selection: {}'.format(n_obs_max))
+                obs_common = obs_common[:n_obs_max]
+
+            obs_table = Table(dict(OBS_ID=obs_common))
+        else:
+            raise ValueError('Invalid selection: {}'.format(selection))
+
+        log.info('Writing {}'.format(self.run_list))
+        obs_table.write(self.run_list, format='ascii.csv')
+
+    def group_observations(self):
+        """Group the background observation list.
+
+        For now we'll just use the zenith binning from HAP.
+        """
+
+        obs_table = self.define_obs_table()
+
+        # Define observation groups
+        # zenith_bins = np.array([0, 20, 30, 40, 50, 90])
+        zenith_bins = np.array([0, 49, 90])
+        # zenith_bins = np.array([0, 30, 90])  # for testing
+        axes = [ObservationGroupAxis('ZEN_PNT', zenith_bins, fmt='edges')]
+        obs_groups = ObservationGroups(axes)
+        log.info(obs_groups.info)
+
+        # Apply observation grouping
+        obs_table = obs_groups.apply(obs_table)
+
+        # Store the results
+        filename = self.obs_table_grouped_filename
+        log.info('Writing {}'.format(filename))
+        obs_table.write(str(filename), format='ascii.ecsv')
+
+        filename = self.group_table_filename
+        log.info('Writing {}'.format(filename))
+        obs_groups.obs_groups_table.write(str(filename), format='ascii.ecsv')
+        self.ntot_group = obs_groups.n_groups
+
+    def make_model(self, modeltype, obs_table=None, excluded_sources=None):
```

Comment:
Not sure to understand this last point…
For the fill_obs method we need the excluded_sources table! You want to directly define the obs_table and the excluded sources as member of the class?

Le 11 mars 2016 à 12:07, Christoph Deil <notifications@github.com> a écrit :

> In gammapy/background/off_data_background_maker.py:
> 
> > +        log.info(obs_groups.info)
> > +
> > +        # Apply observation grouping
> > +        obs_table = obs_groups.apply(obs_table)
> > +
> > +        # Store the results
> > +        filename = self.obs_table_grouped_filename
> > +        log.info('Writing {}'.format(filename))
> > +        obs_table.write(str(filename), format='ascii.ecsv')
> > +
> > +        filename = self.group_table_filename
> > +        log.info('Writing {}'.format(filename))
> > +        obs_groups.obs_groups_table.write(str(filename), format='ascii.ecsv')
> > +        self.ntot_group = obs_groups.n_groups
> > +
> > +    def make_model(self, modeltype, obs_table=None, excluded_sources=None):
> I would remove the obs_table and excluded_sources parameters here and add those earlier, usually in __init__.
> The basic pattern of usage of a maker would then be:
> 
> maker = Maker( ... parameters, inputs, configuration ...)
> maker.run_step1()
> maker.run_step2( option2_parameters )
> I.e. the individual steps would only take a few parameters (if any) ... basically the algorithm is set up in __init__.
> 
> @JouvinLea - What do you think?
> 
> —
> Reply to this email directly or view it on GitHub.
>